### PR TITLE
Bug 1929278: [4.7] Adjust priority class of core platform prometheus

### PR DESCRIPTION
**Pull request by bparees**
* [ ] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: test/e2e/priority_class_test.go [hunk 1]
```diff
@@ -45,4 +50,14 @@ func TestToEnsureUserPriorityClassIsPresentAndLower(t *testing.T) {
 	if userPriorityClass.Value >= systemClusterPriorityClass.Value || userPriorityClass.Value >= systemNodePriorityClass.Value {
 		t.Fatalf("openshift-user-critical was higher priority than existing classes: %d", userPriorityClass.Value)
 	}
+	// Ensure openshift-system-critical is a lower value than the system class' ones, but higher than openshift-user-critical.
+	if openshiftSystemPriorityClass.Value >= systemClusterPriorityClass.Value {
```
Comment (ravisantoshgudimetla) [inline test/e2e/priority_class_test.go:54]: This perhaps is not needed. We any way won't be able to create a priority class greater than system-cluster critical or node critical

File: test/e2e/priority_class_test.go [hunk 2]
```diff
@@ -45,4 +50,14 @@ func TestToEnsureUserPriorityClassIsPresentAndLower(t *testing.T) {
...
+		t.Fatalf("openshift-system-critical cannot be higher than system priority: %d vs %d", openshiftSystemPriorityClass.Value, systemClusterPriorityClass)
+	}
+	if openshiftSystemPriorityClass.Value >= systemNodePriorityClass.Value {
```
Comment (ravisantoshgudimetla) [inline test/e2e/priority_class_test.go:57]: Same as above.

File: test/e2e/priority_class_test.go [hunk 3]
```diff
@@ -45,4 +50,14 @@ func TestToEnsureUserPriorityClassIsPresentAndLower(t *testing.T) {
 	if userPriorityClass.Value >= systemClusterPriorityClass.Value || userPriorityClass.Value >= systemNodePriorityClass.Value {
 		t.Fatalf("openshift-user-critical was higher priority than existing classes: %d", userPriorityClass.Value)
 	}
+	// Ensure openshift-system-critical is a lower value than the system class' ones, but higher than openshift-user-critical.
+	if openshiftSystemPriorityClass.Value >= systemClusterPriorityClass.Value {
+		t.Fatalf("openshift-system-critical cannot be higher than system priority: %d vs %d", openshiftSystemPriorityClass.Value, systemClusterPriorityClass)
```
Comment (ravisantoshgudimetla) [inline test/e2e/priority_class_test.go:55]: We need to change it to `systemClusterPriorityClass.Value`

**Comment by rphillips** [comment]
.Value missing from the Fatalf messages

File: test/e2e/priority_class_test.go (hunk 1 above)
Comment (bparees) [reply test/e2e/priority_class_test.go:54]: someone can revise it later.

**Comment by rphillips** [comment]
```
 CONT  TestMultinamespacePrometheusRule
    client.go:308: unexpected status code response, want 200, got 503 
```
/retest

**Comment by bparees** [comment]
well this is failing the upgrade test due to:
```
fail [github.com/openshift/origin/test/e2e/upgrade/upgrade.go:143]: during upgrade to registry.build01.ci.openshift.org/ci-op-r6wgzl2d/release@sha256:eb8d6f885f1eca3654d6ac3fa7dc285b4686f8dec30634babfef89e4272c0e94
Unexpected error:
    <*errors.errorString | 0xc00246aa50>: {
        s: "Cluster did not complete upgrade: timed out waiting for the condition: Could not update priorityclass \"openshift-user-critical\" (325 of 669): the object is invalid, possibly due to local cluster configuration",
    }
    Cluster did not complete upgrade: timed out waiting for the condition: Could not update priorityclass "openshift-user-critical" (325 of 669): the object is invalid, possibly due to local cluster configuration
occurred
```

which seems to imply we can't reduce the user critical priority value.  So at best we can just set both to 1000000(however many zeroes)

**Comment by bparees** [comment]
priorityclasses can't be updated, so closing this PR

**Pull request closed without merging**